app/models/person.rb

```diff
@@ -33,4 +27,27 @@ class Person < ApplicationRecord
             if: -> { new_record? || !password_confirmation.nil? }
 
   has_secure_password
+
+  scope :from_omniauth, ->(auth) do
+    where(provider: auth.provider, uid: auth.uid)
+  end
+
+  def self.omniauth_retrieval_or_save(auth)
+    from_omniauth(auth).first_or_initialize.tap do |user|
+      user.save_from_omniauth(auth)
+    end
+  end
+
+  def save_from_omniauth(auth)
```

Comment:
This method doesnt belong here. This should be in 'auth/:provider/callback' controller action. It doesnt have to redirect to the same sessions#create action as when a user signs up from our site, instead of with google auth